src/js/libgeo/Prover.js

```diff
@@ -68,21 +74,77 @@ function incidentPC(p, c) {
             var erg = General.mult(c.matrix, p.homog);
             erg = General.mult(p.homog, erg);
             erg = CSNumber.abs(erg);
-            if (erg.value.real < 0.0000000000001) {
-                p.incidences.push(c.name);
-                c.incidences.push(p.name);
-            }
+            return (erg.value.real < 0.0000000000001);
         }
     };
 }
 
 function checkConjectures() {
+    var debug = false;
+    if (debug) console.log("conjectures", conjectures.length);
+    //if (!debug)
     if (conjectures.length === 0) return;
-    // TODO: we need some randomized proving here:
-    // move free elements and check conjectures a number of times.
-    // For now assume that all conjectures could be verified.
+    backupGeo();
```

Comment:
I don't think this is the appropriate function here. `backupGeo` and `restoreGeo` don't maintain a stack, but just a single backup for the sake of animations. So while I see no problem running the prover at widget load time, we will run into trouble if we use those functions while some animation is playing.

But we don't really need `backupGeo`. Much of its code is related to saving and restoring the state of behavior properties. We are only dealing with geometry, here, so copying the geometric state should be sufficient. Add a name like `"prover"` to `stateArrayNames`, then use `stateArrays.prover.set(stateIn);` to back up and `stateIn.set(stateArrays.prover); recalcAll();` to restore.